src/async_fuse/memfs/inode.rs

```diff
@@ -0,0 +1,116 @@
+use std::sync::Arc;
+use clippy_utilities::OverflowArithmetic;
+use tokio::sync::Mutex;
+use crate::async_fuse::fuse::protocol::INum;
+use crate::async_fuse::memfs::{dist, S3MetaData};
+use crate::async_fuse::memfs::dist::etcd;
+use crate::async_fuse::memfs::s3_wrapper::S3BackEnd;
+use crate::common::etcd_delegate::EtcdDelegate;
+
+#[derive(Debug)]
+pub(crate) struct InodeState {
+    range_begin_end:Mutex<(INum,INum)>,
+    recycle_unused:crossbeam_queue::SegQueue<INum>,
+}
+
+impl InodeState {
+    pub fn new()->Self{
+        Self{
+            range_begin_end:Mutex::new((0,0)),
```

Comment:
Use sync Mutex if possible. For example, parking_lot mutex.

Reply:
This lock will be held when fetching range value in etcd.

Reply:
Is that necessary? Usually we only hold IO resource lock (async mutex) while doing IO operations.